browser/ui/webui/brave_education/getting_started_helper.cc

```diff
@@ -0,0 +1,52 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/webui/brave_education/getting_started_helper.h"
+
+#include <utility>
+
+#include "brave/components/brave_education/common/features.h"
+
+namespace brave_education {
+
+namespace {
+
+std::optional<EducationPageType> GetEducationPageType() {
+  if (base::FeatureList::IsEnabled(features::kShowGettingStartedPage)) {
+    return EducationPageType::kGettingStarted;
+  }
+  return std::nullopt;
+}
+
+}  // namespace
+
+GettingStartedHelper::GettingStartedHelper(Profile* profile)
+    : server_checker_(profile) {}
+
+GettingStartedHelper::~GettingStartedHelper() = default;
+
+void GettingStartedHelper::GetEducationURL(GetEducationURLCallback callback) {
+  auto page_type = GetEducationPageType();
+  if (!page_type) {
+    std::move(callback).Run(std::nullopt);
+    return;
+  }
+
+  server_checker_.IsServerPageAvailable(
+      *page_type, base::BindOnce(&GettingStartedHelper::OnCheckerResult,
+                                 base::Unretained(this), std::move(callback)));
```

Comment:
I've looked through the memory allocation here and am reasonably convinced this shouldn't be an issue. With that said, it would be ideal to use a weak_ptr here instead just in case the network calls produce some sort of odd lifetime we haven't thought of or planned for that could introduce a UAF issue.

Reply:
Updated to use a weak ptr.

Reply:
Thank you!